tests/schema.cpp

```diff
@@ -338,7 +338,19 @@ TEST_CASE("Schema") {
                     {"link_to_top_level_object", PropertyType::Object|PropertyType::Nullable, "TopLevelObject"}
                 }}
             };
-            REQUIRE_THROWS_CONTAINING(schema.validate(), "Cycles containing embedded objects are not currently supported: 'EmbeddedObject.link_to_top_level_object.link_to_embedded_object'");
+            REQUIRE_NOTHROW(schema.validate());
+        }
+
+        SECTION("does not reject a top level loop via embedded object link") {
+            Schema schema = {
+                {"TopLevelObject", {
+                    {"link_to_self", PropertyType::Object|PropertyType::Nullable, "TopLevelObject"}
+                }},
+                {"EmbeddedObject", ObjectSchema::IsEmbedded{true}, {
```

Comment:
Nothing is linking to this embedded object, so that schema is probably invalid (we have a ticket to detect freestanding embedded objects, but that hasn't been completed yet).

Reply:
so the server doesn't actually outright reject freestanding embedded object schemas, it will be stored as an "unbound embedded schema" in memory as long as the connection is open, and will be persisted once something links to it and makes it a "bound" embedded schema

Reply:
Right, my point was that it would likely be an error on the developer's end to include a freestanding embedded object in their schema, as there would be nothing meaningful they can do with it. Obviously, it doesn't matter much in a test case, but it would probably be a good idea to use realistic schemas in tests. For example, for this one, I would probably add a link to the embedded object in `TopLevelObject` to verify that both cycles `parent->parent` and `parent->embedded->parent` are allowed.

Reply:
Thanks for the clarifications both! I've updated this particular test schema to be more realistic.